plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServerUtils.java

```diff
@@ -67,57 +72,44 @@
 	public static final String OPENSHIFT_SERVER_TYPE = "org.jboss.tools.openshift.openshift.server.type";//$NON-NLS-1$
 
 	public static final String SERVER_START_ON_CREATION = "org.jboss.tools.openshift.SERVER_START_ON_CREATION";
+
+	public static IServer findServerForService(String serviceName) {
+		final IServerType serverType = getServerType();
+		final Optional<IServer> match = Stream.of(ServerCore.getServers())
+				.filter(server -> server.getServerType()
+						.equals(serverType)
+						&& server.getAttribute(OpenShiftServerUtils.ATTR_SERVICE, "").equals(serviceName))
+				.findAny();
```

Comment:
you can also write something like `.findFirst().orElse(null)` to retrieve the value and avoid the if/else block below.